UIforETW/TraceLoggingSupport.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "stdafx.h"
+#include "TraceLoggingSupport.h"
+#include "Utility.h"
+
+#include <bcrypt.h>
+
+#include <vector>
+#include <array>
+
+#ifdef _DEBUG
+#define new DEBUG_NEW
+#endif
+
+#pragma comment(lib, "Bcrypt.lib")
+
+static std::system_error MakeSystemErrorFromWin32Error(DWORD errorCode, const char* context)
```

Comment:
I added an exception handler in the Starttracing. I can see LCMapStringEx failing for some invalid strings, but I wasn't able to trivially cause that, and if MFC is going to catch that exception silently, then that would defeat the goal of the fail fast.

I also squashed the commits.

Mildly on topic, do you know if this issue https://randomascii.wordpress.com/2012/07/05/when-even-crashing-doesnt-work/ still applies to recent Windows versions?